var/spack/repos/builtin/packages/py-warpx/package.py

```diff
@@ -40,20 +41,25 @@ class PyWarpx(PythonPackage):
     variant('mpi', default=True,
             description='Enable MPI support')
 
-    for v in ['21.12', '21.11', '21.10', '21.09', '21.08', '21.07', '21.06',
-              '21.05', '21.04', 'develop']:
+    for v in ['22.01',
+              '21.12', '21.11', '21.10', '21.09', '21.08', '21.07', '21.06',
+              '21.05', '21.04',
+              'develop']:
         depends_on('warpx@{0}'.format(v),
                    when='@{0}'.format(v),
                    type=['build', 'link'])
 
-    depends_on('python@3.6:3.9', type=('build', 'run'))
+    depends_on('python@3.6:3.9', type=('build', 'run'), when='@:21.12')
+    depends_on('python@3.6:3.10', type=('build', 'run'), when='@22.01:')
```

Comment:
Actually even more relaxed now:
https://github.com/ECP-WarpX/WarpX/pull/2664
```suggestion
    depends_on('python@3.6:', type=('build', 'run'), when='@22.01:')
```